go.mod

```diff
@@ -1,6 +1,6 @@
 module go.kubeguard.dev/guard
 
-go 1.18
+go 1.23.1
```

Comment:
this updates minimum version to be go 1.23.1. By just building with latest go, you'll get rid of the stdlib CVEs. Unless the project is using go 1.23 features, you don't need to bump this.

Reply:
1.23 has already been used in [CI](https://github.com/kubeguard/guard/pull/394/files), so I think to align all golang version should be ok?